### ISSUE TITLE
`super` keyword in static class property initializer is not compiled correctly

### ISSUE THREAD

**Issue by MichaelMitchell-at** [evanw/esbuild#2108]
```js
class Foo {
    static items = []
}

class Bar extends Foo {
    static items = super.items.concat([])
}
```

gets compiled to
```js
class Foo {
}
Foo.items = [];
class Bar extends Foo {
}
Bar.items = super.items.concat([]);
```

Both [tsc](https://www.typescriptlang.org/play?target=3#code/MYGwhgzhAEBiD29oG8CwAoaXoQC5lwEthpDcBTAWxgF5oBtAXQwF8MNRIYAhMAJ2jkAHhQB2AExgIkaTNjwFipCtWh0IAVwAO5PgDoyVCHuDxRwAgAomASlZA) and [Babel](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAEBiD29oG8BQ0PQgFzNglsNPtgKYC2MAvNANoC6qAvqqqJDAEJgBO0pADzIA7ACYwESNJiy4CREhWpYArgAdSPAHSLKW4PGHA8ACgYBKZkA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-2&prettier=false&targets=&version=7.17.7&externalPlugins=&assumptions=%7B%7D) seem to correctly compile away the `super` keyword

**Comment by evanw** [comment]
I can't reproduce this. I assume you're using TypeScript given how class fields are transformed into assignments, which only happens with TypeScript. I get this output with the current version of esbuild:

```js
var __getProtoOf = Object.getPrototypeOf;
var __reflectGet = Reflect.get;
var __reflectSet = Reflect.set;
var __superStaticGet = (obj, member) => __reflectGet(__getProtoOf(obj), member, obj);
class Foo {
}
Foo.items = [];
const _Bar = class extends Foo {
};
let Bar = _Bar;
Bar.items = __superStaticGet(_Bar, "items").concat([]);
```

Are you using esbuild 0.13.10 or older? A bug like this was fixed in esbuild 0.13.11, which was released back in October of last year. If the bug still happens with the latest version of esbuild, then please provide instructions for how to reproduce it.

**Comment by MichaelMitchell-at** [comment]
Ah, we're on `v0.12.13`. Will try to upgrade then, thanks.
